src/action_cc.cc

```diff
@@ -295,7 +295,50 @@ NM_ACTION_(nickel_setting) {
 NM_ACTION_(nickel_extras) {
     #define NM_ERR_RET nullptr
 
-    if (!strcmp(arg, "web_browser")) {
+    if (!strncmp(arg, "web_browser", 11)) {
+        bool modal;
+        QUrl *url;
+        QString *css;
+
+        if (!strcmp(arg, "web_browser")) {
+            modal = false;
+            url = new QUrl();
+            css = new QString("");
+        } else {
+            char *tmp1 = strdupa(arg); // strsep and strtrim will modify it
+            char *arg1 = strtrim(strsep(&tmp1, ":"));
+            char *arg2 = strtrim(tmp1);
+
+            if (!arg2 || strcmp(arg1, "web_browser"))
+                NM_RETURN_ERR("unknown beta feature name or plugin mimetype '%s' (split: '%s')", arg, arg1);
+
+            QString tmp = QString::fromUtf8(arg2).trimmed();
+
+            if (tmp.section(':', 0, 0).trimmed() == "modal") {
+                modal = true;
+                tmp = tmp.section(':', 1).trimmed();
+            } else {
+                modal = false;
+            }
```

Comment:
```suggestion
        bool modal = false;
        QUrl *url;
        QString *css;

        if (!strcmp(arg, "web_browser")) {
            url = new QUrl();
            css = new QString("");
        } else {
            char *tmp1 = strdupa(arg); // strsep and strtrim will modify it
            char *arg1 = strtrim(strsep(&tmp1, ":"));
            char *arg2 = strtrim(tmp1);

            if (!arg2 || strcmp(arg1, "web_browser"))
                NM_RETURN_ERR("unknown beta feature name or plugin mimetype '%s' (split: '%s')", arg, arg1);

            QString tmp = QString::fromUtf8(arg2).trimmed();

            if (tmp.section(':', 0, 0).trimmed() == "modal") {
                modal = true;
                tmp = tmp.section(':', 1).trimmed();
            }
```
May as well set the default modal bool to false. Saves a few of lines of code.

I'm still new to this code suggestion thingy, so I may not have done this right...